### ISSUE TITLE
Remove swap in activities

### ISSUE THREAD

**Issue by zhangzhongnan928** [AlphaWallet/alpha-wallet-ios#2902]

**Comment by hboon** [comment]
Let’s put it as Contract Execution until we work out a better way. Maybe build a list of contracts fused by a few popular DEXes sites.

**Comment by hboon** [comment]
(Or until we display it as describe above)

**Comment by zhangzhongnan928** [comment]
> (Or until we display it as describe above)

is this possible👇? 

Before TokenScript is adopted, I suggest using a standard way to show all the activities titles:
If the contract code is verified on Etherscan, show the activity title as Contract(or DApp) "contract name(contract address if no name)" function "xxxxx"
If the contract code is not verified on Etherscan, show the activity title as Contract(or DApp) "contract name(contract address if no name)" function

**Comment by hboon** [comment]
Yes, it sounds like it is indeed possible to get the ABI (and hence we can derive the function signature and check against transactions) for verified contracts — https://etherscan.io/apis#contracts.

Good idea. I'll keep this up so Tomek has a chance to look at this and feedback + design when he's back. I'll also look into whether we can build a list of swap contracts and function calls. Maybe it'll be a good way to get Swap displays more accurate and more useful.

I'll do a quick patch and replace the "Swap" with "Contract Execution" first for the imminent release first. #2903

**Comment by colourfreak** [comment]
Hard to say how to implement it. I can't answer this question. 

Prototype:
https://invis.io/KU114ACT6GCT

This is the prototype with all types of activities we could have. There are many more. Bug this is a good start. A regular user does not know what is 'contract execution'. So that information is even more mysterious in opposition to not having this at all. 

The design is ready. Waiting for a long time to be developed. What else should I provide to replace a development process? There is no substitute for hard work sometimes. Maybe dig in how other wallets like Zerion and Rainbow implement it? 

If Tokenscript is limiting us with this subject, let's build it from scratch not using TokenScript. But that requires hard work again.

**Comment by hboon** [comment]
It’ll come. Swap label has already removed, just not released yet. Is the design up on Zeplin?

**Comment by colourfreak** [comment]
https://zpl.io/bzd0Zol
